lib/canonicalization.ts

```diff
@@ -58,17 +58,14 @@ export async function computeCanonicalDataset(state: GlobalState, input: InputDa
     // The input to the algorithm can be either an nQuads document, or a dataset
     // representation with Quads. This function makes the nQuad document "disappear" from
     // the rest of the processing.
-    const convertToQuads = (inp: InputDataset): Quads => {
-        if (typeof inp === 'string') {
-            return parseNquads(inp as string);
-        } else if (Array.isArray(inp)) {
-            return new Set<rdf.Quad>(inp as rdf.Quad[]);
-        } else {
-            return inp;
-        }
-    };
-
-    const input_dataset: Quads = convertToQuads(input);
+    // const convertToQuads = (inp: InputDataset): InputQuads => {
+    //     if (typeof inp === 'string') {
+    //         return parseNquads(inp as string);
+    //     } else {
+    //         return inp;
+    //     }
+    // };
+    const input_dataset: InputQuads = (typeof input === 'string') ? parseNquads(input as string) : input;
     const retval: Quads = new Set();
```

Comment:
If there are cases where your algorithm can output identical quads on some cases then you'll want to make this an N3 Store as well and add quads to it using the `store.add` method.

Reply:
Yes, I was wondering about that, but I believe there is no such case in the algorithm. Ie, the Set is o.k.